### PR TITLE
Patch golang in kube-cross with CL38131 for ppc64le platform

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:
Faced golang issues([Issue 19425](https://github.com/golang/go/issues/19425)) while building latest kubernetes on ppc64le environment with golang1.8 version and go fixed part of [CL 38131](https://go-review.googlesource.com/c/38131/). I'm patching golang1.8 to avoid any issues while cross building the kubernetes.
**Which issue this PR fixes** 
**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by ixdy** [comment]
Is the upstream fix likely to land in a future go1.8 release?

**Comment by mkumatag** [comment]
@laboger is working on backporting it to 1.8

File: build/build-image/cross/Dockerfile [hunk 1]
```diff
@@ -79,10 +79,15 @@ RUN export ETCD_VERSION=v3.0.17; \
...
 
-# Prebuild the standard library for those platforms that need the edge GOROOT
-RUN for platform in linux/arm linux/ppc64le; do GOOS=${platform%/*} GOARCH=${platform##*/} GOROOT=${K8S_EDGE_GOROOT} go install std; done
+# We need a patched go1.8 for linux/ppc64le (https://github.com/golang/go/issues/19425)
+COPY golang-patches/CL38131-go1.8.patch ${K8S_EDGE_GOROOT}/
+RUN cd ${K8S_EDGE_GOROOT} \
+  && patch -p1 < CL38131-go1.8.patch \
+  && cd src \
+  && GOROOT_FINAL=${K8S_EDGE_GOROOT} GOROOT_BOOTSTRAP=/usr/local/go ./make.bash \
+  && for platform in linux/arm linux/ppc64le; do GOOS=${platform%/*} GOARCH=${platform##*/} GOROOT=${K8S_EDGE_GOROOT} go install std; done
```
Comment (ixdy) [inline build/build-image/cross/Dockerfile]: do we need to do this for `linux/arm` too? if so, update the comment on L87. If not, remove it here?

Comment (mkumatag) [inline build/build-image/cross/Dockerfile]: This patch is only for ppc64le, L93 is nothing to do with this patch, I'm just keeping it is as from previous version L88

Comment (mkumatag) [inline build/build-image/cross/Dockerfile]: @luxas do we still need to build standard library even after we built whole golang?

**Comment by ixdy** [comment]
LGTM. I will build a new cross build image closer to the end of code freeze when this PR is likely to merge.

File: build/build-image/cross/Dockerfile (hunk 1 above)
Comment (luxas) [inline build/build-image/cross/Dockerfile]: Yes, it will speed up all the other builds if std is precompiled.

**Review by luxas — APPROVED** [review]
I'm a little bit scared of doing changes like this just before a release, but I guess we don't have any choice here really.

@ixdy can you push the image soon so I at least can verify it doesn't break anything before we merge this?

**Comment by luxas** [comment]
cc @laboger does this affect arm as well or is this a ppc64le-only thing?

And how come didn't our crossbuild detect this and start failing?
The cross-build CI seems to be healthy, but the description in https://github.com/golang/go/issues/19425 shows it should error at compile-time.

**Comment by ixdy** [comment]
Does this need to merge into 1.6?

**Comment by mkumatag** [comment]
@ixdy I don't want to take chance here... Let's push it to 1.6 as well...

**Comment by ixdy** [comment]
I'm not sure if this is too late to include in 1.6.0. /cc @kubernetes/release-team 

@luxas I'm likewise confused why this isn't failing our crossbuild - maybe it's something that broke between 1.8rc2 and 1.8?

**Comment by luxas** [comment]
@ixdy I looked into it and now I see what the problem is. `net` and `os/user` fail in some circumstances on ppc64le and maybe arm.

However, it seems like it doesn't affect the server and client binaries we're including in the cross-build, but some other tools like `genfeddocs`.

I think we should wait with this one, hope that https://go-review.googlesource.com/c/38131/ is cherrypicked into go1.8.1 and then bump to that go version for ppc64le and arm in time for v1.6.1 of kubernetes.

cc @ixdy @bradfitz @cherrymui sounds like a plan?

**Comment by mkumatag** [comment]
Created https://github.com/kubernetes/kubernetes/pull/44216 for moving K8S_GOLANG_EDGE_VERSION to go1.8.1

**Comment by ixdy** [comment]
We probably don't need this one anymore, right?

**Comment by mkumatag** [comment]
Yes.. you are right...

**Pull request closed without merging**